optuna/integration/_lightgbm_tuner/optimize.py

```diff
@@ -225,11 +225,11 @@ def _preprocess(self, trial: optuna.trial.Trial) -> None:
             self.lgbm_params["bagging_fraction"] = param_value
         if "bagging_freq" in self.target_param_names:
             self.lgbm_params["bagging_freq"] = trial.suggest_int("bagging_freq", 1, 7)
-        if "min_child_samples" in self.target_param_names:
-            # `GridSampler` is used for sampling min_child_samples value.
+        if "min_data_in_leaf" in self.target_param_names:
+            # `GridSampler` is used for sampling min_data_in_leaf value.
             # The value 1.0 for the hyperparameter is always sampled.
-            param_value = trial.suggest_int("min_child_samples", 5, 100)
-            self.lgbm_params["min_child_samples"] = param_value
+            param_value = trial.suggest_int("min_data_in_leaf", 5, 100)
+            self.lgbm_params["min_data_in_leaf"] = param_value
```

Comment:
Could you copy the value of `min_data_in_leaf` value to `min_child_samples` here? If we copy it in the post-processing, the `min_child_samples` value is empty in the best booster.

```suggestion
            self.lgbm_params["min_data_in_leaf"] = param_value

        if "min_data_in_leaf" in self.lgbm_params:
            self.lgbm_params["min_child_samples"] = self.lgbm_params["min_data_in_leaf"]
```

Reply:
In addition, please add a test case to ensure that the optimized parameters have `min_child_samples`.